standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/DatabaseProduct.java

```diff
@@ -749,6 +753,22 @@ public Object getBoolean(boolean val) {
     return val;
   }
 
+  public Object convertDateValue(Object dateValue) {
+    assert dateValue instanceof String;
+    Date date = MetaStoreUtils.convertStringToDate((String)dateValue);
+    Object result = MetaStoreUtils.convertDateToString(date);
+    return result;
+  }
+
+  public Object convertTimestampValue(Object timestampValue) {
+    assert timestampValue instanceof String;
+    MetaStoreUtils.convertStringToTimestamp((String) timestampValue);
```

Comment:
what is this? are we mutating the input object?

Reply:
To check the literal `timestampValue` is a valid timestamp

Reply:
same as above, it's not a convert method - `isValidTimestamp(String timestampValue)`